bootloader/grub_test.go

```diff
@@ -53,6 +56,15 @@ func (s *grubTestSuite) SetUpTest(c *C) {
 	bootloader.MockGrubFiles(c, s.rootdir)
 
 	s.bootdir = filepath.Join(s.rootdir, "boot")
+	// By default assume amd64 in the tests: there are specialized
+	// tests for other arches
+	s.realArch = arch.DpkgArchitecture()
+	arch.SetArchitecture("amd64")
```

Comment:
you could use 

```
s.AddCleanup(archtest.MockArchitecture("amd64"))
```

here and avoid the TearDownTest need

Reply:
Nice one, I've changed this now